napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -48,8 +51,28 @@
 )
 from napari.utils.translations import trans
 
+# TODO: add error icon and handle pip install errors
+
+
+def is_conda_package(pkg):
+    """Determines if plugin was installed through conda.
+
+    Returns
+    -------
+    bool: True if a conda package, False if not
+    """
+
+    # Installed conda packages within a conda installation and environment can be identified as files
+    # with the template `<package-name>-<version>-<build-string>.json` saved within a `conda-meta` folder within
+    # the given environment of interest.
+    conda_meta_dir = Path(sys.prefix) / 'conda-meta'
+	return any(True for _ in conda_meta_dir.glob(f"{name}-*-*.json"))
```

Comment:
Today I realize my solution is not as good as the previous one. 
In my case, for package `napari`, it will return true even if `napari` is not installed from conda, but `napari-foobar` is installed from conda. 

Here is a sample solution, but it uses `re` module so I'm not sure if it is still correct. 

```suggestion
    return any(re.match(rf"{name}-[^-]+-[^-]+.json", p.name) for p in conda_meta_dir.glob(f"{name}-*-*.json"))
```